### DISCUSSION TITLE
Slow PSR-12 analysis

### DISCUSSION THREAD

**Discussion by nc1vr** [PHP-CS-Fixer/PHP-CS-Fixer#8096] [Q&A] [answered]
Is it possible to reduce the time using PSR-12? I have a large PHP file of 5630 lines and it takes about 40 sec to complete analysis. Switching to deprecated PSR-2 takes 3.8 sec only and doesn't matter if Parallel Runner On or Off.

**Comment by keradus** [comment, marked as answer]
Hey @aimi617 , thanks for your feedback !

We do our best to provide the top speed for Fixer, which includes:
- [parallel execution](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.59.3/doc/usage.rst#the-fix-command) - for case you described you see no difference, as parallel execution works on files, and you have a single file
- [cache - for consecutive executions](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.59.3/doc/usage.rst#caching) - so next time you run the Fixer, it will skip the file if not changed.

The more rules you use, it will require more CPU to compute the analysis (ie, PSR12 has more rules than PSR2, hence the time difference between running those sets).

It would be indeed fantastic to see even faster execution.

With your description, it is not possible to reproduce the situation. I strongly encourage you to contribute: debug the situation on your side and raise PR with improvement.
If you are not feeling up for it - provide a fully reproducible scenario (fixer version, PHP version, config file, command you use, file of 5630 lines) to make it easier for ppl to invest their personal time to improve situation for your individual case. Thanks!
